### ISSUE TITLE
Hello, is that possible support windows or mac?

### ISSUE THREAD

**Issue by lucasjinreal** [rhasspy/piper#45]
Many users like me wanna have a local client on windows, it would be very useful if it can support build on windows!

(I would suggest let users download onnxruntime from official themself and then link it)

**Comment by clockworkwhale** [comment]
I got it working in WSL (Ubuntu 22.04) under Windows 11. There seems to be little or no speed loss running it that way too.

But yeah, a native Windows version would be easier to work with. If rhasspy isn't a Windows guy then obviously it's totally fair enough for him not to bother though. Let's just hope this repo gets some more attention since that'll for sure lead to someone making a Windows fork or PR.

It's weird how little notice it's getting. I've been searching for a truly realtime TTS solution that sounds better than espeak for a long time, and this appears to finally be it.  Most TTS projects that claim realtime performance are exaggerating or just lying, but inference with piper is so fast I can hardly believe it.

**Comment by lucasjinreal** [comment]
@clockworkwhale are u testing on GPU or CPU?
I hope rhasspy can help support windows since there are some lib it looks hard to build on windows like portaudio, this lib even using ancient autogen and make.

**Comment by clockworkwhale** [comment]
> @clockworkwhale are u testing on GPU or CPU? I hope rhasspy can help support windows since there are some lib it looks hard to build on windows like portaudio, this lib even using ancient autogen and make.

CPU (Ryzen 5950x). It's fast enough already on CPU that there's no point to using GPU, it's like 200milliseconds to generate 10 seconds of speech lol.

**Comment by Wetzel402** [comment]
I would just use WSL or Docker.

**Comment by lucasjinreal** [comment]
@t-mat  can u help: \piper\espeak-ng-data\phontab': No such file or directory.?

**Comment by t-mat** [comment]
No, I can't.  Because I do not understand how you've got that error.

**Comment by lucasjinreal** [comment]
@t-mat  Please, you downloaded ng data there is no such hontab, do u get this phontab file under easpak-ng folder>?>????????

**Comment by lucasjinreal** [comment]
@t-mat Also, your espeack init didn't try to found this file?????? Are we using same espack-ng??

**Comment by lucasjinreal** [comment]
@t-mat  can u tell me how to get phantab file???????

**Comment by t-mat** [comment]
As for espeak-ng, Read the README.md
https://github.com/rhasspy/piper#installation

see the following lines
https://github.com/rhasspy/piper/blob/283bf2851823579a67334bc61456c2c53b06bd01/Dockerfile#L16-L30

and this directory
https://github.com/rhasspy/piper/tree/283bf2851823579a67334bc61456c2c53b06bd01/lib

**Comment by lucasjinreal** [comment]
@t-mat what u paste didn't make sense to me, I still didn't found phontab .......
I am using your script and I download the espack-ngpatched.zip, unziped it, didn't found this file....................

**Comment by t-mat** [comment]
I and anyone didn't download `espeak-ng-data/phon*` files.  `espeak-ng` generates them.
I'm not sure you've actually read this repository and run my script or not.   But anyway, the following command generates them.

```
cd \path\to\espeak-ng
cd espeak-ng\src\windows
.\x64\Release\espeak-ng.exe --path=..\.. --compile-phonemes
dir ..\..\espeak-ng-data\
```

**Comment by notBradPitt** [comment]
@t-mat Thanks for sharing such a comprehensive build script. However, I gotta confess that I still ran into some troubles (and I reckon it's got more to do with my system than your code), so I was thinking if it'd possible for you to share your compiled build instead? That'd really save a lot of headaches here 😅

**Comment by lucasjinreal** [comment]
I tried, but the result is horrible......
is that normal?
[1684592873311876800.zip](https://github.com/rhasspy/piper/files/11522526/1684592873311876800.zip)

**Comment by synesthesiam** [comment]
A Windows pre-release is available here for testing: https://github.com/rhasspy/piper/releases/tag/2023.9.9-1

**Comment by clockworkwhale** [comment]
> A Windows pre-release is available here for testing: https://github.com/rhasspy/piper/releases/tag/2023.9.9-1

Tested in Windows 11 on a 5950x and it's working fantastic! Many thanks for this. Speed on CPU appears to be just as insanely fast as Linux, i.e. essentially instant generation even on the larger models. ~200 ms to generate several seconds of speech.

**Comment by synesthesiam** [comment]
Thanks for testing, @clockworkwhale!

**Comment by lucasjinreal** [comment]
@clockworkwhale hello, is there any steps build on my own side?

Also, does Chinese works?

**Comment by clockworkwhale** [comment]
> @clockworkwhale hello, is there any steps build on my own side?

No need to build, as synesthesiam has provided a prebuilt executable in the release he linked to.

I'm afraid I only tested English. However, Chinese voice models appear to be available on the HuggingFace page here: https://huggingface.co/rhasspy/piper-voices/tree/v1.0.0/zh

**Comment by notBradPitt** [comment]
I've tested the Windows build and it is beautiful! Piper still runs amazingly fast even when using an old CPU and the result are almost perfect. Hands down one of the best TTS I've used!

**Comment by RoxBlox3** [comment]
Hello how do i start Piper on Windows. I downloaded the Windows release and unzipped it. When i tried to open the Piper app it opens cmd and closes it immediately do i need to download something else first?

**Comment by sjc57** [comment]
What is the correct syntax for running on windows command line?
It tried   echo "hello" | piper --model en_US-lessac-high.onnx --output_file hello.wav   but get unsupported 16-bit application popup window  (I'm rinning 64 bit Windows 10)
Thanks
